### PR TITLE
Add user account functions

### PR DESCRIPTION
**Pull request by keharper**
This PR adds functions for user account, except those that pertain to order history.

### CODE REVIEW

**Review by KostiantynFandeliuk — APPROVED** [review]

File: src/content/docs/dropins/user-account/functions.mdx [hunk 1]
```diff
@@ -0,0 +1,578 @@
+---
+title: User account functions
+description: Learn about the API functions provided by the User Account dropin.
+---
+
+import OptionsTable from '@components/OptionsTable.astro';
+
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx:7]: Consider adding a short lead-in sentence like [cart](https://experienceleague.adobe.com/developer/commerce/storefront/dropins/cart/cart-functions/) and [PDP](https://experienceleague.adobe.com/developer/commerce/storefront/dropins/product-details/pdp-functions/).

File: src/content/docs/dropins/user-account/functions.mdx [hunk 2]
```diff
@@ -0,0 +1,578 @@
+---
+title: User account functions
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx]:
```suggestion
title: User Account functions
```

File: src/content/docs/dropins/user-account/functions.mdx [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+  street: '123 Main St',
+  streetMultiline_2: 'Suite 100',
+  defaultShipping: true,
+  defaultBilling: true,
+};
+```
+
+## getAttributesForm
+
+The `getAttributesForm` function uses the [`attributesForm`](https://developer.adobe.com/commerce/webapi/graphql/schema/attributes/queries/attributes-form/) query to retrieve EAV attributes associated with customer and customer address frontend forms. The `formCode` parameter must be one of the following values: `customer_account_create`, `customer_account_edit`, `customer_address_create`, or `customer_address_edit`.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx:83]: Should we use a bulleted list after 2 or 3 items?

File: src/content/docs/dropins/user-account/functions.mdx [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+
+The `getCustomer` function retrieves the customer information for the logged-in customer. The function uses the [`customer`](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/queries/customer/) query.
+
+```ts
+export const getCustomer = async (): Promise<CustomerDataModelShort>
+```
+
+### Returns
+
+Returns a promise that resolves to CustomerDataModelShort object:
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx]:
```suggestion
Returns a promise that resolves to `CustomerDataModelShort` object:
```

File: src/content/docs/dropins/user-account/functions.mdx [hunk 5]
```diff
@@ -0,0 +1,578 @@
...
+
+```javascript
+import { getCustomerAddress } from '@/account/api/getCustomerAddress';
+
+getCustomerAddress();
+```
+
+## getOrderHistoryList
+
+The `getOrderHistoryList` function is an asynchronous function that retrieves a list of customer orders using the [`customer`](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/queries/customer/) query. . It optionally takes parameters for pagination and filtering.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx]:
```suggestion
The `getOrderHistoryList` function is an asynchronous function that retrieves a list of customer orders using the [`customer`](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/queries/customer/) query. It optionally takes parameters for pagination and filtering.
```

File: src/content/docs/dropins/user-account/functions.mdx [hunk 6]
```diff
@@ -0,0 +1,578 @@
...
+  selectOrdersDate: string,
+  currentPage: number
+): Promise<OrderHistory | null>
+```
+
+<OptionsTable
+  compact
+  options={[
+    ['Parameter', 'Type', 'Req?', 'Description'],
+    ['pageSize', 'number', 'No', 'the maximum number of results to return at once. The default value is 20.'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx]:
```suggestion
    ['pageSize', 'number', 'No', 'The maximum number of results to return at once. The default value is 20.'],
```

File: src/content/docs/dropins/user-account/functions.mdx [hunk 7]
```diff
@@ -0,0 +1,578 @@
...
+export const getRegions = async (
+  countryCode: string
+): Promise<RegionTransform[]
+```
+
+<OptionsTable
+  compact
+  options={[
+    ['Parameter', 'Type', 'Req?', 'Description'],
+    ['countryCode', 'string', 'Yes', 'a two-letter abbreviation for the country ID'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx]:
```suggestion
    ['countryCode', 'string', 'Yes', 'A two-letter abbreviation for the country ID.'],
```

File: src/content/docs/dropins/user-account/functions.mdx [hunk 8]
```diff
@@ -0,0 +1,578 @@
...
+import { updateCustomer } from '@/account/api/updateCustomer';
+
+updateCustomer(form: CustomerUpdateInput);
+```
+
+## updateCustomerAddress
+
+The `updateCustomerAddress` function updates an address associated with the current customer. The function uses the [`updateCustomerAddress`](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/update-address/) mutation.
+
+The `forms` object includes an `addressId`, Aa number representing the ID of the address to be updated and other address details as defined in CustomerAddressesModel.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-account/functions.mdx]:
```suggestion
The `forms` object includes an `addressId`, which is a number representing the ID of the address to be updated and other address details as defined in `CustomerAddressesModel`.
```

**Review by jeff-matthews — COMMENTED** [review]
Just few minor editorial suggestions

**Review by jeff-matthews — APPROVED** [review]

**Pull request merged**